4 implement state management strategy

Menubrea opened the pull request:
- Picked Zustand for State Management
- Installed Zustand
- Created `store` folder with an a basic exampleStore.
- Created component making use of exampleStore.
- Created Selector function for easier for cleaner import usage.

If anyone have any reservations on using `Zustand`, or would prefer a different library for ´State-management` and/or if there is anything specific you would like implemented at this point in time, please let me know.

Menubrea committed:
Installed zustand

Menubrea committed:
Created exampleStore

Menubrea committed:
Created example component using exampleStore

Menubrea committed:
Created selector

Menubrea committed:
Created barrel export and configured example files to use updated imports with selectors

Menubrea committed:
Destructured Store in example

Moesera reviewed (approved):
Looks like a good starter point. I dont have anything that i want to add for now. I like working with Zustand so i'm fine with the choice.

kyrregjerstad reviewed (approved):
Looks good to me!

Menubrea committed:
Added example dispatch with variable value, changed exampleStore type definition and reset state  to better match official documentation and practices

Menubrea committed:
changed exampleStore to exampleSlice, with updated syntax for global storage hook

Menubrea committed:
Created global store hook and associated import/export script files

Menubrea commented:
**Updates** 
- Changed structure to modular `state slices` to be imported to be accessed by global state hook. 
- Added documentation for how to use.

Let me know what you think about the updated structure. 

```tsx
import { useStore } from '~/store'

// Destructured
const { count, increment } = useStore();
// alternatively using dot notation, which will expose all available properties as a list.
let count = useStore().count;
```

kyrregjerstad commented:
I'm a fan of the Destructured method

Anclagen commented:
Looks good to me, I think Zustand will server are purposes for a small scale app.
One thing I was wondering is will we need to use something like [SWR](https://swr.vercel.app/) or [React Query](https://www.npmjs.com/package/@tanstack/react-query) for caching progress server side, although we can just update the database as checks are complete too, but might be something for consideration.

Menubrea commented:
> Looks good to me, I think Zustand will server are purposes for a small scale app. One thing I was wondering is will we need to use something like [SWR](https://swr.vercel.app/) or [React Query](https://www.npmjs.com/package/@tanstack/react-query) for caching progress server side, although we can just update the database as checks are complete too, but might be something for consideration.

Absolutely, great input! Luckily for us [tPRC's queries](https://trpc.io/docs/client/react) functions as a wrapper around [React Query](https://www.npmjs.com/package/@tanstack/react-query) hooks. For any server data, I anticipate queries will be our go-to for caching and state-management.
